File: q50-q59/q59/q59.py
class Solution:
    def generateMatrix(self, n: int) -> List[List[int]]:
        solution = [[0 for x in range(n)] for y in range(n)]
        count = 1
        layer = 0
        for layer in range(0, int ((n + 1)/2)):
            for ptr1 in range(layer, n - layer -1):
                solution[layer][ptr1] = count
                count += 1
            for ptr2 in range(layer, n - layer - 1):
                solution[ptr2][n - layer - 1] = count
                count += 1
            for ptr3 in range(n - layer -1, layer, -1):
                solution[n - layer - 1][ptr3] = count
                count += 1
            for ptr4 in range(n - layer - 1, layer, -1):
                solution[ptr4][layer] = count
                count += 1
        if n % 2:
            solution[int(n/2)][int(n/2)] = count
        return solution

File: Sequence/q1-q99/q50-q59/q59/q59.cpp
class Solution {
public:
    vector<vector<int>> generateMatrix(int n) {
        vector<vector<int>> solution (n, vector<int>(n));
        int count = 1;
        int layer;
        for (layer = 0; layer < (n + 1)/2; layer++){
            for (int ptr1 = layer; ptr1 < n - layer - 1; ptr1++){
                solution[layer][ptr1] = count++;
            }
            for (int ptr2 = layer; ptr2 < n - layer - 1; ptr2++){
                solution[ptr2][n - layer - 1] = count++;
            }
            for (int ptr3 = n - layer - 1; ptr3 > layer; ptr3--){
                solution[n - layer - 1][ptr3] = count++;
            }
            for (int ptr4 = n - layer - 1; ptr4 > layer; ptr4--){
                solution[ptr4][layer] = count++;
            }
        }
        if (n % 2)
            solution[n/2][n/2] = count;
        return solution;
    }
};
